src/esp/geo/VoxelGrid.cpp

```diff
@@ -0,0 +1,340 @@
+  addGrid<bool>("Boundary");
+  auto boundaryGrid = getGrid<bool>("Boundary");
+
+  int num_filled = 0;
+  // Transfer data from Volume to VoxelGrid
+  for (int i = 0; i < dims[0]; i++) {
+    for (int j = 0; j < dims[1]; j++) {
+      for (int k = 0; k < dims[2]; k++) {
+        if (vhacdVolume->GetVoxel(i, j, k) >= 2) {
+          num_filled++;
+          boundaryGrid[i][j][k] = true;
+        }
+      }
+    }
+  }
+  if (num_filled == dims[0] * dims[1] * dims[2]) {
+    // When VHACD is given too low of a resolution
+    Mn::Debug() << "VOXELIZATION FAILED";
+  }
+}
+#endif
+
+VoxelGrid::VoxelGrid(const Mn::Vector3& voxelSize,
+                     const Mn::Vector3i& voxelGridDimensions) {
+  m_voxelSize = voxelSize;
+  m_voxelGridDimensions = voxelGridDimensions;
+  m_offset = Mn::Vector3(0.0, 0.0, 0.0);
+  addGrid<bool>("Boundary");
+}
+
+template <>
+VoxelGridType VoxelGrid::voxelGridTypeFor<bool>() {
+  return VoxelGridType::Bool;
+}
+template <>
+VoxelGridType VoxelGrid::voxelGridTypeFor<int>() {
+  return VoxelGridType::Int;
+}
+template <>
+VoxelGridType VoxelGrid::voxelGridTypeFor<float>() {
+  return VoxelGridType::Float;
+}
+template <>
+VoxelGridType VoxelGrid::voxelGridTypeFor<Mn::Vector3>() {
+  return VoxelGridType::Vector3;
+}
+
+//  --== GETTERS AND SETTERS FOR VOXELS ==--
+
+std::vector<std::pair<std::string, std::string>> VoxelGrid::getExistingGrids() {
+  std::vector<std::pair<std::string, std::string>> existingGrids =
+      std::vector<std::pair<std::string, std::string>>();
+  std::map<std::string, GridEntry>::iterator it;
+  for (it = grids_.begin(); it != grids_.end(); it++) {
+    std::string typeName;
+    VoxelGridType type = it->second.type;
+    if (type == VoxelGridType::Bool) {
+      typeName = "Bool";
+    } else if (type == VoxelGridType::Int) {
+      typeName = "Int";
+    } else if (type == VoxelGridType::Float) {
+      typeName = "Float";
+    } else if (type == VoxelGridType::Vector3) {
+      typeName = "Vector3";
+    }
+    existingGrids.push_back(std::make_pair(it->first, typeName));
+  }
+  return existingGrids;
+}
+
+std::shared_ptr<Mn::Trade::MeshData> VoxelGrid::getMeshData(
+    const std::string& gridName) {
+  if (meshDataDict_[gridName] == nullptr)
+    generateMesh(gridName);
+  return meshDataDict_[gridName];
+}
+
+Mn::GL::Mesh& VoxelGrid::getMeshGL(const std::string& gridName) {
+  if (meshDataDict_[gridName] == nullptr)
+    generateMesh(gridName);
+  return meshGLDict_[gridName];
+}
+
+Mn::Vector3 VoxelGrid::getGlobalCoords(const Mn::Vector3i& coords) {
+  return Mn::Vector3(coords) * m_voxelSize + m_offset;
+}
+
+void VoxelGrid::fillBoolGridNeighborhood(std::vector<bool>& neighbors,
+                                         const std::string& gridName,
+                                         const Mn::Vector3i& index) {
+  std::vector<Mn::Vector3i> increments = {{0, 0, 1},  {1, 0, 0},  {0, 1, 0},
+                                          {0, 0, -1}, {0, -1, 0}, {-1, 0, 0}};
```

Comment:
Can be just `Mn::Vector3i increments[] = {...}`, no need for a vector. Furthermore, it can be `constexpr`.